cpp/log/etcd_consistent_store-inl.h

```diff
@@ -72,10 +72,12 @@ EtcdConsistentStore<Logged>::NextAvailableSequenceNumber() const {
     return sequenced_entries.back().Entry().sequence_number() + 1;
   }
 
-  // TODO(alcutter): Implement the rest of the logic around /serving_sth too
-  // once there are methods to inspect that.
-  LOG(WARNING) << "NextAvailableSequenceNumber() not checking /serving_sth.";
-  return 0;
+  if (!serving_sth_) {
+    LOG(WARNING) << "Log has no Serving STH [new log?], returning 0";
```

Comment:
I wasn't 100% sure, but wouldn't returning -1 and having the signer not do anything when that happens be safer?

Even in the single node case, it should elect itself leader, then pick a serving STH based on its local database tree size (which would make a lot more sense than 0!)... If it did sign something using 0, while the database had sequenced entries in it, it would be Very Bad (tm).